src/etc/inc/certs.inc

```diff
@@ -622,6 +647,40 @@ function cert_get_publickey($str_crt, $decode = true, $type = "crt") {
 			break;
 	}
 	unlink($certfn);
+	return ($out !== false ? implode("\n", $out) : false);
+}
+
+function cert_encrypt_key($str_key, $password, $decode = true) {
+	if ($decode) {
+		$str_key = base64_decode($str_key);
+	}
+
+	$keyfn = tempnam('/tmp', 'CEK');
+	$passfn = tempnam('/tmp', 'CEK');
+	file_put_contents($keyfn, $str_key);
+	file_put_contents($passfn, $password);
+
+	exec("/usr/bin/openssl pkey -in {$keyfn} -des3 -passout file:{$passfn} 2>/dev/null", $out);
```

Comment:
Why 3DES (`-des3`) here, and not something more secure, like `-aes128` or `-aes256`?

Reply:
I just always used `-des3`, I don't have a preference either way.

Reply:
I'd prefer to see this as at least `-aes128`, otherwise this looks OK to merge and test.